gobcore/database/reader/objectstore.py

```diff
@@ -25,10 +25,14 @@ def query_objectstore(connection, config):
     for row in result:
         if pattern.match(row["name"]):
             file_info = dict(row)    # File information
-            if file_type == "XLS":
-                # Include (non-empty) Excel rows
+            if file_type in ["XLS", "CSV"]:
                 obj = get_object(connection, row, container_name)
-                for item in _read_xls(obj, file_info):
+                if file_type == "XLS":
+                    # Include (non-empty) Excel rows
+                    _read = _read_xls
+                else:
+                    _read = _read_csv
+                for item in _read(obj, file_info, config):
```

Comment:
This loops seems unnecessary, as both _read_xls and _read_csv already return a generator.

` return _read(obj, file_info, config)`

should do the trick.

Reply:
Nice, had that in the code but didn’t work. Will try again, thanks